File: Leetcode 0680 Valid Palindrome II [easy] [string]/Valid Palindrome II [two-pointers].java
class Solution {
    public boolean validPalindrome(String s) {
        int start = 0;
        int end = s.length() - 1;
        while (start < end) {
            if (s.charAt(start) == s.charAt(end)) {
                start++;
                end--;
            } else {
                return helper(s, start + 1, end) || helper(s, start, end - 1);
            }
        }
        
        return true;
    }
    
    private boolean helper(String s, int i, int j) {
        int start = i;
        int end = j;
        while (start < end) {
            if (s.charAt(start) == s.charAt(end)) {
                start++;
                end--;
            } else {
                return false;
            }
        }
        
        return true;
    }
}

File: Leetcode 0680 Valid Palindrome II [easy] [string]/Valid Palindrome II [two-pointers].js
/**
 * @param {string} s
 * @return {boolean}
 */
var validPalindrome = function(s) {
    let start = 0;
    let end = s.length - 1;
    while (start < end) {
        if (s[start] === s[end]) {
            start++;
            end--;
        } else {
            return helper(s, start + 1, end) || helper(s, start, end - 1);
        }
    }
    
    return true;
};

let helper = (s, i, j) => {
    let start = i;
    let end = j;
    while (start < end) {
        if (s[start] === s[end]) {
            start++;
            end--;
        } else {
            return false;   
        }
    }
    
    return true;
}